working/v3.0.0-draft2/ciba-flow/CIBA-authorization-request-response-schema.json

```diff
@@ -0,0 +1,32 @@
+{
+  "definitions": {},
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://apicentre.paymentsnz.co.nz/schemas/CIBA-authorization-request-response-schema.json",
+  "type": "object",
+  "title": "The schema for the CIBA request response object (OIDC/FAPI compliant)",
+  "additionalProperties": false,
+  "required": ["auth_req_id", "expires_in"],
+  "properties": {
+    "auth_req_id": {
+      "$id": "/properties/auth_req_id",
+      "type": "string",
+      "description": "This is a unique identifier for the authentication request made by the CIBA Client (third party)",
+      "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
```

Comment:
we can specify "format": "uuid"

Reply:
The OIDC spec doesn't mention this is required to be a UUID, so this pattern is overly restrictive.

From https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0-ID1.html:

> auth_req_id
> REQUIRED. This is a unique identifier to identify the authentication request made by the Client. It MUST contain sufficient entropy (a minimum of 128 bits while 160 bits is recommended) to make brute force guessing or forgery of a valid auth_req_id computationally infeasible - the means of achieving this are implementation specific, with possible approaches including secure pseudorandom number generation or cryptographically secured self-contained tokens.

Reply:
@jamesmccann agreed UUID is overly restrictive, and the API centre have an open issue about this also.  It was actually raised by me over a year ago.  The early drafts of CIBA all showed UUID values which I think is the reason this crept in, but that is not a reason to keep it.  Lets discuss at TWG.

Reply:
As per the CIBA description of `auth_req_id`:

> auth_req_id
> REQUIRED. This is a unique identifier to identify the authentication request made by the Client. It MUST contain sufficient entropy (a minimum of 128 bits while 160 bits is recommended) to make brute force guessing or forgery of a valid auth_req_id computationally infeasible - the means of achieving this are implementation-specific, with possible approaches including secure pseudorandom number generation or cryptographically secured self-contained tokens. The OpenID Provider MUST restrict the characters used to 'A'-'Z', 'a'-'z', '0'-'9', '.', '-' and '_', to reduce the chance of the client incorrectly decoding or re-encoding the auth_req_id; this character set was chosen to allow the server to use unpadded base64url if it wishes. The identifier MUST be treated as opaque by the client.

Adding a regex to match this pattern, and minLength of 22 (since this is the minimum number of characters for the 128 bit entropy in base64url encoding)